### ISSUE TITLE
Replace SHA3-224 algorithm in `text-buffer` by SHA1

### ISSUE THREAD

**Issue by Akirathan** [enso-org/enso#8662]
In `text-buffer`, SHA3-224 is used for calculating the checksum of a text content - https://github.com/enso-org/enso/blob/689c8f7c3c9bb745d8d65e35384a22f084fd2160/lib/scala/text-buffer/src/main/scala/org/enso/text/Sha3_224VersionCalculator.scala#L11

According to the original author, @4e6, there is no reason to use specifically SHA3-224, so we can just use the simpler SHA-1.

Note that this change also requires modifications to some gui parts.

Related to:
- #8362 
- #8553
- #8664

**Comment by hubertp** [comment]
A duplicate of https://github.com/enso-org/enso/issues/6691 I think.

**Comment by 4e6** [comment]
I think the scope is different. This one only implies the change of hashing algorithm in the `text/applyEdit` request.

**Comment by 4e6** [comment]
Also @Akirathan, if the goal is just to remove the bouncycastle dependency, can the version calclulator use the same logic with the `MessageDigest` from the standard library as `FileSystem` does? Without changing the algo
https://github.com/enso-org/enso/blob/689c8f7c3c9bb745d8d65e35384a22f084fd2160/engine/language-server/src/main/scala/org/enso/languageserver/filemanager/FileSystem.scala#L234-L248

**Comment by 4e6** [comment]
Ok, I see, you already did it in #8664

**Comment by Akirathan** [comment]
> Ok, I see, you already did it in #8664

Yes. Removing the `org.bouncycastle` dependency is a related, but different task. And it is done in #8664.

**Comment by hubertp** [comment]
> I think the scope is different. This one only implies the change of hashing algorithm in the `text/applyEdit` request.

Fine, but there is a problem (at least currently) when we have to support both, old and new, GUI. They can't use different algorithms. I would say this would have to wait until old GUI is completely dropped and still will be an incompatible change.

**Comment by JaroslavTulach** [comment]
> this would have to wait until old GUI is completely dropped

Or we can fix the old GUI to use SHA-1. I assume downgrading the algorithm is going to be easy.

**Comment by jdunkerley** [comment]
On hold until GUI1 is retired then we will switch to SHA1 in GUI2.

**Comment by JaroslavTulach** [comment]
Anybody interested in implementing this one? If so, please reopen.
